spec/features/projects/life_cycle/overview_page/sidebar_spec.rb

```diff
@@ -0,0 +1,153 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require_relative "shared_context"
+
+RSpec.describe "Show project life cycles on project overview page", :js, :with_cuprite, with_flag: { stages_and_gates: true } do
+  include_context "with seeded projects and stages and gates"
+
+  let(:overview_page) { Pages::Projects::Show.new(project) }
+
+  before do
+    login_as admin
+  end
+
+  it "does show the sidebar" do
+    overview_page.visit_page
+    overview_page.expect_visible_sidebar
+  end
+
+  context "when stages and gates are disabled", with_flag: { stages_and_gates: false } do
+    it "does not show the sidebar" do
+      overview_page.visit_page
+      overview_page.expect_no_visible_sidebar
+    end
+  end
+
+  context "when all stages and gates are disabled for this project" do
+    before do
+      project_life_cycles.each { |p| p.toggle!(:active) }
+    end
+
+    it "does not show the sidebar" do
+      overview_page.visit_page
+      overview_page.expect_no_visible_sidebar
+    end
+  end
+
+  describe "with correct order and scoping" do
+    it "shows the project stages and gates in the correct order" do
+      overview_page.visit_page
+
+      overview_page.within_life_cycles_sidebar do
+        fields = page.all(".op-project-life-cycle-container")
+
+        expect(fields.size).to eq(7)
+
+        expect(fields[0].text).to include("Initiating")
+        expect(fields[1].text).to include("Ready for Planning")
+        expect(fields[2].text).to include("Planning")
+        expect(fields[3].text).to include("Ready for Executing")
+        expect(fields[4].text).to include("Executing")
+        expect(fields[5].text).to include("Ready for Closing")
+        expect(fields[6].text).to include("Closing")
+      end
```

Comment:
This could be simplified:

```suggestion
    overview_page.within_life_cycles_sidebar do
      expected_stages = [
        "Initiating",
        "Ready for Planning",
        "Planning",
        "Ready for Executing",
        "Executing",
        "Ready for Closing",
        "Closing"
      ]

      fields = page.all(".op-project-life-cycle-container")
      expect(fields.map(&:text)).to eq(expected_stages)
    end
```